docs/proposals/11-snapshots.md

```diff
@@ -0,0 +1,191 @@
+enhancement to the `Volume` resource to provide a path for restoring from a snapshot. These new resources and 
+changes would provide users with increased capabilities for data protection and disaster recovery.
+
+## Background
+
+The `onmetal-api` project currently provides interfaces for managing resources such as `Volumes` and `VolumePools`. 
+A `VolumePool` is a representation of a `VolumePool` provider that materializes the requested `Volume` in an underlying
+storage implementation. However, there is no direct way to snapshot the content of a `Volume`, nor is there a method 
+for defining different types of snapshots or restoring from a snapshot, which potentially puts users' data at risk.
+
+## Goals
+
+1. **Introduce `Snapshot` Resource**: Enable users to take point-in-time snapshots of the content of a `Volume`.
+2. **Introduce `SnapshotClass` Resource**: Define the different types of snapshots that a `VolumePool` can support.
+3. **Implement Restoration Path**: Enhance the `Volume` resource to include a `dataSource` field, providing a path for 
+restoring data from a `Snapshot`.
+4. **Improve Data Protection**: By creating snapshots and allowing data restoration, users can protect their data and 
+restore it in case of loss or corruption.
+5. **Support for Different Types of Snapshots**: With the introduction of `SnapshotClass`, support for different types 
+of snapshots based on `VolumePool` capabilities will be provided.
+6. **Enhance Flexibility**: Provide users with the ability to choose the type of snapshot they want to use based on 
+their specific needs.
+
+## Non-Goals
+
+1. **Modify Existing `VolumePool` Resource**: This proposal aims to introduce new resources and an enhancement to 
+the `Volume` resource and doesn't intend to modify the behavior of the existing `VolumePool` resources.
+3. **Provide Backup or Archive Functionality**: The `Snapshot` resource will facilitate data protection through 
+snapshots, but it's not designed to provide a complete backup or archival solution for `Volumes`.
+4. **Ensure Immediate Availability of Snapshot**: While snapshots aim to provide point-in-time data protection, 
+the actual creation and readiness of a snapshot can depend on various factors such as `Volume` size, 
+`VolumePool` capabilities, and infrastructure performance.
+5. **Replace Existing Disaster Recovery Solutions**: The `Snapshot` resource and restoration path are meant to be 
+additional tools in the data protection toolbox, not to replace existing comprehensive disaster recovery and 
+data protection solutions.
+
+## API Design
+
+### `Snapshot` Resource
+
+The `Snapshot` resource will have the following structure:
+
+```yaml
+apiVersion: storage.api.onmetal.de/v1alpha1
+kind: Snapshot
+metadata:
+  name: example-snapshot
+spec:
+  volumeRef:
+    name: example-volume
+  snapshotClassRef:
+    name: example-snapshotclass
```

Comment:
Can we also have encryption support for Snapshot as Volume encryption support is there?

Reply:
I would expect a transitive behavior here. If the volume is encrypted, the snapshot should be encrypted as well. Plus, the option to encrypt the snapshot explicitly.